src/PresentationalComponents/TagsToolbar/TagsToolbar.js

```diff
@@ -12,29 +12,47 @@ import PropTypes from 'prop-types';
 import TagIcon from '@patternfly/react-icons/dist/js/icons/tag-icon';
 import { addNotification } from '@redhat-cloud-services/frontend-components-notifications';
 import { connect } from 'react-redux';
-import { injectIntl } from 'react-intl';
+import intersection from 'lodash/intersection';
 import messages from '../../Messages';
 import { setSelectedTags } from '../../AppActions';
+import { tagUrlBuilder } from './Common';
+import { useIntl } from 'react-intl';
 
-const TagsToolbar = ({ selectedTags, intl, setSelectedTags }) => {
+const TagsToolbar = ({ selectedTags, setSelectedTags }) => {
+    const intl = useIntl();
     const [isOpen, setIsOpen] = useState(false);
     const [tags, setTags] = useState([]);
     const [manageTagsModalOpen, setManageTagsModalOpen] = useState(false);
     const showMoreCount = 20;
     const onToggle = isOpen => setIsOpen(isOpen);
 
     useEffect(() => {
+        const url = new URL(window.location);
+        let params = new URLSearchParams(url.search);
+        params = params.get('tags');
         const fetchTags = async () => {
             try {
                 const response = await API.get(`${BASE_URL}/tag/`);
                 setTags(response.data.tags);
+                if (params.length) {
+                    const tagsToSet = intersection(params.split(','), response.data.tags);
+                    tagUrlBuilder(tagsToSet);
+                    setSelectedTags(tagsToSet);
+                }
             } catch (error) {
                 addNotification({ variant: 'danger', dismissable: true, title: intl.formatMessage(messages.error), description: `${error}` });
             }
         };
 
         fetchTags();
-    }, [intl]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        console.error(window.location.search, selectedTags);
```

Comment:
Do you want to console.error these values?

Reply:
OOOO GOOD POINT